alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportGetFindingsAction.kt

```diff
@@ -131,10 +137,34 @@ class TransportGetFindingsSearchAction @Inject constructor(
         }
     }
 
-    suspend fun search(searchSourceBuilder: SearchSourceBuilder): GetFindingsResponse {
+    suspend fun resolveFindingsIndexName(findingsRequest: GetFindingsRequest): String {
+        var indexName = ""
+
+        if (findingsRequest.findingIndexName.isNullOrEmpty() == false) {
+            // findingIndexName has highest priority, so use that if available
+            indexName = findingsRequest.findingIndexName
+        } else if (findingsRequest.monitorId != null) {
+            // second best is monitorId.
+            // We will use it to fetch monitor and then read indexName from dataSources field of monitor
+            withContext(Dispatchers.IO) {
+                val getMonitorRequest = GetMonitorRequest(findingsRequest.monitorId, -3L, RestRequest.Method.GET, null)
+                val getMonitorResponse: GetMonitorResponse =
```

Comment:
we should catch exceptiton if GetmonitorAction fails and fail the get findings action overall?

Reply:
This should be happening on line 134 right now. I'll add IT for this